lib/egaugex.ex

```diff
@@ -12,9 +12,10 @@ defmodule Egaugex do
   def egauge_parser(egauge_id, opts \\ []) do
     username = opts[:username] || nil
     password = opts[:password] || nil
+    base_url = opts[:base_url] || nil
     uri = opts[:uri] || "/cgi-bin/egauge-show?S&n=60"
     realm = opts[:realm] || "eGauge Administration"
-    result = Egaugex.Fetcher.get_egauge_data(egauge_id, username, password, uri, realm)
+    result = Egaugex.Fetcher.get_egauge_data(egauge_id, username, password, uri, realm, base_url)
```

Comment:
At some point we can just pass the opts obj to eGauge and use guards to then add the auth header :)